pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1416,20 +1417,34 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(claim *v1.Persis
 	}
 
 	var pluginName string
+	provisionerName := storageClass.Provisioner
 	if plugin != nil {
-		pluginName = plugin.GetPluginName()
+		if plugin.IsMigratedToCSI() {
```

Comment:
Please add an unit test for this. `"11-17 - external provisioner"` could be a good start. You need to inject a volume plugin that's migrated to CSI, some variant of `wrapTestWithPluginCalls` might be sufficient.

Reply:
Updated PR with `11-21 - external provisioner for CSI migration` testcase to make sure for CSI migration scenarios, provisioning ends up waiting for a claim with annotation `vendor.com/MockCSIPlugin` and event `Normal ExternalProvisioning`.